plugins/woocommerce/src/Blocks/BlockTypes/AddToCartWithOptions.php

```diff
@@ -68,12 +68,16 @@ protected function render( $attributes, $content, $block ) {
 
 		$previous_product = $product;
 		$product          = wc_get_product( $post_id );
-		if ( ! $product instanceof \WC_Product ) {
+		if ( ! $product instanceof \WC_Product || ( ProductType::SIMPLE === $product->get_type() && ! $product->is_purchasable() ) ) {
 			$product = $previous_product;
 
 			return '';
 		}
 
+		if ( ! $product->is_in_stock() ) {
+			return wc_get_stock_html( $product );
```

Comment:
Should we restore the previous product here before returning?

Reply:
Ups, true. Good catch! Fixed in e4acf4c2f59c4861f3b70ed5d45bce84c7d81a59.